lib/web_ui/lib/src/engine/picture.dart

```diff
@@ -54,6 +54,11 @@ class EnginePicture implements ui.Picture {
       ..src = imageDataUrl
       ..width = width
       ..height = height;
+
+    // The image loads asynchronously. We need to wait before returning,
+    // otherwise the returned HtmlImage will be temporarily unusable.
+    await imageElement.onLoad.first;
```

Comment:
needs to handle failure.

Reply:
if fails, it could leak. i would write very explicit code here with handler add/remove

Reply:
Done.